src/gui_common/charts/line/LineChart.cs

```diff
@@ -577,7 +577,8 @@ private void CreateDropDownLegend(string title)
     /// </summary>
     private void RenderChart()
     {
-        if (dataSets == null || dataSets.Count <= 0)
+        // Handle empty or entirely hidden datasets
+        if (dataSets?.All(d => !d.Value.Draw) == true)
```

Comment:
It may be my own lack of knowledge of C# and x?. structures speaking here, but why use ==true here?

Reply:
Pls use != false , because when dataset == null it was supposed to draw empty string, but now null == true returns false.

Reply:
Yeah, just experienced that on my own PR. Then I'm fine with the code!

Reply:
> It may be my own lack of knowledge of C# and x?. structures speaking here, but why use ==true here?

The ? syntax is conditional calling of the method. If the variable in front of the ? is null it won't call the method afterwards and instead automatically return null. Because Any returns a bool, which is a primitive type and therefore non nullable it automatically returns a nullable version of bool instead (`bool?`). A nullable bool can have three states: true, false or null. Thats why explicit == true is required here. Note: because of this == true is different than != false, which can be quite confusing

Reply:
So why not using !=false ? Should use here

Reply:
Good catch, just tested and it wasn't working anymore for null cases. Requested changes for that...